src/test/java/org/candlepin/model/ContentAccessPayloadTest.java

```diff
@@ -0,0 +1,203 @@
+        ContentAccessPayload payload = new ContentAccessPayload();
+
+        assertThrows(IllegalArgumentException.class, () -> {
+            payload.setOwner(owner);
+        });
+    }
+
+    @Test
+    public void testSetOwner() {
+        String expectedOwnerId = TestUtil.randomString("id-");
+        Owner owner = new Owner()
+            .setId(expectedOwnerId);
+
+        ContentAccessPayload payload = new ContentAccessPayload()
+            .setOwner(owner);
+
+        assertThat(payload.getOwnerId())
+            .isNotNull()
+            .isEqualTo(expectedOwnerId);
+    }
+
+    @Test
+    public void testSetOwnerIdWithNullValue() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            new ContentAccessPayload()
+                .setOwnerId(null);
+        });
+    }
+
+    @Test
+    public void testSetOwnerId() {
+        String expectedOwnerId = TestUtil.randomString("id-");
+
+        ContentAccessPayload payload = new ContentAccessPayload()
+            .setOwnerId(expectedOwnerId);
+
+        assertThat(payload.getOwnerId())
+            .isNotNull()
+            .isEqualTo(expectedOwnerId);
+    }
+
+    @Test
+    public void testSetPayloadKeyWithNullValue() {
+        ContentAccessPayload payload = new ContentAccessPayload();
+
+        assertThrows(IllegalArgumentException.class, () -> {
+            payload.setPayloadKey(null);
+        });
+    }
+
+    @Test
+    public void testSetPayloadKey() {
+        String expectedPayloadKey = TestUtil.randomString("key-");
+
+        ContentAccessPayload payload = new ContentAccessPayload()
+            .setPayloadKey(expectedPayloadKey);
+
+        assertThat(payload.getPayloadKey())
+            .isNotNull()
+            .isEqualTo(expectedPayloadKey);
+    }
+
+    @Test
+    public void testSetPayloadWithNullValue() {
+        ContentAccessPayload payload = new ContentAccessPayload();
+
+        assertThrows(IllegalArgumentException.class, () -> {
+            payload.setPayload(null);
+        });
+    }
+
+    @Test
+    public void testSetPayload() {
+        String expectedPayload = TestUtil.randomString("payload-");
+
+        ContentAccessPayload payload = new ContentAccessPayload()
+            .setPayload(expectedPayload);
+
+        assertThat(payload.getPayload())
+            .isNotNull()
+            .isEqualTo(expectedPayload);
+    }
+
+    @Test
+    public void testSetTimestampWithNullValue() {
+        ContentAccessPayload payload = new ContentAccessPayload();
+
+        assertThrows(IllegalArgumentException.class, () -> {
+            payload.setTimestamp(null);
+        });
+    }
+
+    @Test
+    public void testSetTimestamp() {
+        Date expected = new Date();
+
+        ContentAccessPayload payload = new ContentAccessPayload()
+            .setTimestamp(expected);
+
+        assertThat(payload.getTimestamp())
+            .isNotNull()
+            .isEqualTo(expected);
+    }
+
+    @Test
+    public void testToString() {
```

Comment:
I'm iffy about adding tests for toString since we don't use this as anything other than debug output. Maybe as a quick verification that it can't return null or otherwise break, but I'm not sure we need to (nor want to) verify the format. Thoughts?

Reply:
I agree. We should at least verify that the we are generating a string value and don't error out. It does become annoying to maintain a test that verifies the exact string value as time goes on and fields are added/removed.